identity/identity_recovery_test.go

```diff
@@ -12,7 +12,7 @@ func TestNewRecoveryEmailAddress(t *testing.T) {
 	iid := x.NewUUID()
 	a := NewRecoveryEmailAddress("foo@ory.sh", iid)
 
-	assert.Equal(t, a.Value, "foo@ory.sh")
-	assert.Equal(t, a.Via, RecoveryAddressTypeEmail)
-	assert.NotEmpty(t, a.ID)
+	assert.Equal(t, "foo@ory.sh", a.Value)
+	assert.Equal(t, RecoveryAddressTypeEmail, a.Via)
+	assert.Equal(t, iid, a.IdentityID)
```

Comment:
Should we not keep the test for `a.ID`? You can do e.g. `assert.NotEqual(t, uuid.Nil, a.ID)`

Reply:
That's the issue though, it _is_ Equal. `uuid.Nil` is in essence an array of length 16 of 0. And that's what is set into the struct here, as the ID isn't set by `NewRecoveryEmailAddress(...)`.

Reply:
I've now also fixed some places where the check was now failing. 

The issue boils down to the following:
By default, nothing is set into the ID fields of the struct when calling their `New...` functions. This means, that they're populated with the default value for `uuid.UUID` which is, in essence, a byte array with 16 elements, all of which are `0x00`. Now the `assert.NotEmpty` check was changed from looking only at the length of array types to determine whether their empty to comparing them to the default value of the declared type. If this comparison yields `true` the test case fails. 

This is the commit in `stretchr/testify` that introduced this behaviour: https://github.com/stretchr/testify/commit/840cb801497147a6d30a568e0874dfefb10867a9

Reply:
I see, in that case it should be assert.Equal(t, uuid.Nil, id) :)

Reply:
I‘d say that would be testing language features, but the check shouldn’t hurt, so I‘ll add it tomorrow. :) 
Thanks for your patience :)

Reply:
I added a `assert.Equal(t, uuid.Nil, id)` check. Please have another look, thanks :)

Reply:
It doesn't looks like the assertin ohas been added above, can you please check?

Reply:
I also just realised that the original assertion was `NotEmpty` while it actually should have been `assert.Empty`, which probably didn't help with confusion. Sorry about that, that git blame is on me ;)